packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -141,11 +106,112 @@ export const BlockSwitcherDropdownMenu = ( { clientIds, blocks } ) => {
 	const hasBlockOrBlockVariationTransforms =
 		hasPossibleBlockTransformations ||
 		hasPossibleBlockVariationTransformations;
-	const showDropdown =
+	const hasContents =
 		hasBlockStyles ||
 		hasBlockOrBlockVariationTransforms ||
 		hasPatternTransformation;
-	if ( ! showDropdown ) {
+	if ( ! hasContents ) {
+		return <p>{ __( 'There is no available transform.' ) }</p>;
+	}
+	return (
+		<div className="block-editor-block-switcher__container">
+			{ hasPatternTransformation && (
+				<PatternTransformationsMenu
+					blocks={ blocks }
+					patterns={ patterns }
+					onSelect={ ( transformedBlocks ) => {
+						onPatternTransform( transformedBlocks );
+						onClose();
+					} }
+				/>
+			) }
+			{ hasBlockOrBlockVariationTransforms && (
+				<BlockTransformationsMenu
+					className="block-editor-block-switcher__transforms__menugroup"
+					possibleBlockTransformations={
+						possibleBlockTransformations
+					}
+					possibleBlockVariationTransformations={
+						blockVariationTransformations
+					}
+					blocks={ blocks }
+					onSelect={ ( name ) => {
+						onBlockTransform( name );
+						onClose();
+					} }
+					onSelectVariation={ ( name ) => {
+						onBlockVariationTransform( name );
+						onClose();
+					} }
+				/>
+			) }
+			{ hasBlockStyles && (
+				<BlockStylesMenu
+					hoveredBlock={ blocks[ 0 ] }
+					onSwitch={ onClose }
+				/>
+			) }
+		</div>
+	);
+}
+
+export const BlockSwitcher = ( { clientIds } ) => {
+	const blockInformation = useBlockDisplayInformation( clientIds?.[ 0 ] );
+	const { canRemove, hasBlockStyles, icon, blocks, invalidBlocks } =
```

Comment:
Can `hasStyles` be moved too?

Reply:
The `invalidBlocks` is missing from the returned value in `mapSelect`.

Reply:
It's not a heavy operation I think to keep here and not display a dropdown at all in this case..